src/sentry/api/serializers/models/organization.py

```diff
@@ -100,7 +101,22 @@
 
 class BaseOrganizationSerializer(serializers.Serializer):
     name = serializers.CharField(max_length=64)
-    slug = serializers.RegexField(r"^[a-zA-Z0-9][a-zA-Z0-9-]*(?<!-)$", max_length=50)
+
+    # The slug pattern consists of the following:
+    # (?![0-9]+$)   - Negative lookahead to ensure the slug is not all numbers
+    # [a-zA-Z0-9]   - The slug must start with a letter or number
+    # [a-zA-Z0-9-]* - The slug can contain letters, numbers, and dashes
+    # (?<!-)        - Negative lookbehind to ensure the slug does not end with a dash
+    slug = serializers.RegexField(
```

Comment:
This is the only change in this PR that needs to be flagged. You can have a flagr check here and keep line 103 if that doesn't pass.